code/modules/mob/living/simple_animal/hostile/netherworld.dm

```diff
@@ -47,10 +76,115 @@
 	..()
 	if(stat)
 		return
-	if(prob(10))
+	if(prob(10) && !hushed)
 		var/chosen_sound = pick(migo_sounds)
 		playsound(src, chosen_sound, 100, TRUE)
 
+/mob/living/simple_animal/hostile/netherworld/migo/Logout()
+	..()
+	hushed = FALSE
+	if(impersonation)
+		ChangeVoice()
+
+/mob/living/simple_animal/hostile/netherworld/migo/GetVoice()
+	if(impersonation)
+		return impersonation
+	return "[src]"
+
+/mob/living/simple_animal/hostile/netherworld/migo/proc/Hush()
+	set name = "Hush"
+	set category = "Nether"
+	set desc = "Toggles passive noises and noises when you talk."
+	if(hushed)
+		hushed = FALSE
+		to_chat(src, "<span class='notice'>You will now occasionally mimic noises in this dimension.</span>")
+	else
+		hushed = TRUE
+		to_chat(src, "<span class='notice'>You decide against mimicking noises for now.</span>")
+
+/mob/living/simple_animal/hostile/netherworld/migo/proc/ChangeVoice()//hey, not returning null in this is important! don't @ me!
+	set name = "Tune Voice"
+	set category = "Nether"
+	set desc = "Changes your voice to someone else. You will copy the voice patterns if the name matches them."
+	if(impersonation)
+		if(client)
```

Comment:
You don't really need to check if client exists when sending text to mob, to_chat will handle it.